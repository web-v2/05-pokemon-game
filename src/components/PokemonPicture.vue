<template>
  <div class="pokemon-container">
    <img v-if="!showPokemon" 
         :src="imgSrc" 
         class="hidden-pokemon"
         alt="pokemon">
    <img v-else 
         :src="imgSrc" 
         class="fade-in"
         alt="pokemon">         
  </div>
</template>

<script>
export default {
    props: {        
        pokemonId: {
            type: Number,
            required: true
        },        
        showPokemon:{ 
            type: Boolean,
            required: true,
            default: false
        }
    },
    data() {
        return {
            Id: (this.pokemonId <= 0)?1:this.pokemonId
        }
    },
    computed:{        
        imgSrc(){    
            console.log(this.Id);        
            return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.Id}.svg`
        }
    }
}
</script>

<style scoped>
/* Pokemon Picture */
.pokemon-container {
    height: 200px;
}
img {
  height: 200px;
  position: absolute;
  width: 100vw;
  margin-left: -50vw;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.hidden-pokemon {
    filter: brightness(0);
}
</style>