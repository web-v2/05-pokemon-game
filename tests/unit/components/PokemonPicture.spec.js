import {shallowMount} from '@vue/test-utils'
import PokemonPicture from '@/components/PokemonPicture'

describe('PokemonPicture Component', () => {
    test('Debe de hacer match con el snapshot ', () => {
        const wrapper  = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 1,
                showPokemon: true
            }
        })
        expect( wrapper.html() ).toMatchSnapshot()
    });
    test('Debe mostrar la img oculta y el pokemon 100', () => {
        const wrapper  = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: false
            }
        })     
        const [img1, img2] = wrapper.findAll('img')
        expect( img1.exists()).toBeTruthy()
        expect( img2).toBe(undefined)
        expect( img1.classes('hidden-pokemon')).toBeTruthy()

        expect(wrapper.vm.imgSrc).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')        
    });
    test('Debe de mostrar la img si el showPokemon:true', () => {
        const wrapper  = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: true
            }
        })
        const img = wrapper.find('img')        
        expect( img.exists() ).toBeTruthy()
        expect( img.classes('hidden-pokemon') ).toBe(false)  
        expect( img.classes('fade-in') ).toBe(true)  
    });
});