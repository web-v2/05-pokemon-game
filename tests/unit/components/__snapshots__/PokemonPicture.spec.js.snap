// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture Component Debe de hacer match con el snapshot  1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="fade-in" alt="pokemon"></div>`;
